### PR TITLE
Explicit bot types

### PR DESCRIPTION
**Pull request by Celandriel**
## Pull Request Description

This changes how bot type definitions are determined. Instead of lookups, or potentially weak logic, we instead set a flag within the botAI. This eliminates the lookup for players (bots) with botAI. All else fall back to the lookup. 

Some performance details for the current approach.

Metric	Time
min	87 ns
median	7.6 µs
mean	12.5 µs
p90	33 µs
p99	69 µs
max	144 µs

THe min values were when first starting the server, as we loaded more bots this took longer and longer. 

## Feature Evaluation

- Describe the **minimum logic** required to achieve the intended behavior.
- Describe the **processing cost** when this logic executes across many bots.

## How to Test the Changes

Alts, addclass, and randombots still perform as expected. 

## Impact Assessment

- Does this change increase per-bot/per-tick processing or risk scaling poorly with thousands of bots?
    - - [x] No, not at all
    - - [ ] Minimal impact (**explain below**)
    - - [ ] Moderate impact (**explain below**)

- Does this change modify default bot behavior?
    - - [ ] No
    - - [x] Yes (**explain why**)

All bots who are loaded by a player are either alts or addclass bots. This is regardless of whether the bot is from an addclass labled account or not. 

- Does this change add new decision branches or increase maintenance complexity?
    - - [x] No
    - - [ ] Yes (**explain below**)

## AI Assistance

Was AI assistance used while working on this change?
- - [ ] No
- - [x] Yes (**explain below**)

Claude Opus. Implementation. USer reviewed. 

## Final Checklist

- - [x] Stability is not compromised.
- - [x] Performance impact is understood, tested, and acceptable.
- - [x] Added logic complexity is justified and explained.
- - [x] Any new bot dialogue lines are translated.
- - [x] Documentation updated if needed (Conf comments, WiKi commands).

## Notes for Reviewers

In case you missed it before this defines all user summoned bots as either alts, or addclass.

### CODE REVIEW

File: src/Bot/PlayerbotAI.h [hunk 1]
```diff
@@ -230,6 +230,15 @@ enum class BotTypeNumber : uint8
...
+enum class BotAccountType : uint8
+{
+    UNDEFINED = 0,
+    ALT,
+    RANDOMBOT,
+    ADDCLASS,
+    REALPLAYER  // self-bot
+};
+
 enum class GrouperType : uint8
```
Comment (NoxMax) [inline src/Bot/PlayerbotAI.h:242]: Using the term "account" is a bit confusing with the pre-existing system that handles account creations and logins. Not to mention that a single account can have alts, self-bots, and regular player characters. Drop the term entirely, particularly that this is a per-bot related matter.

current | proposed
-- | --
BotAccountType | BotType
_accountType | _botType
GetAccountType() | GetBotType()
SetAccountType() | SetBotType()
BotAccountType::UNDEFINED | BotType::UNDEFINED

Also you accidentally posted to master. You also accidentally created the branch out of master so switching the target to test-staging would bring the merge commits from master 😅 but it should be fine.

File: src/Bot/PlayerbotMgr.cpp [hunk 2]
```diff
@@ -1632,6 +1633,13 @@ void PlayerbotMgr::OnBotLoginInternal(Player* const bot)
     botAI->SetMaster(master);
     botAI->ResetStrategies();
 
+    // Any bot summoned off a randombot account (random pool or addclass) is
+    // treated as an addclass bot; only a player's own characters are alts.
+    uint32 botAccountId = bot->GetSession()->GetAccountId();
+    botAI->SetAccountType(sPlayerbotAIConfig.IsInRandomAccountList(botAccountId)
+                              ? BotAccountType::ADDCLASS
+                              : BotAccountType::ALT);
```
Comment (kadeshar) [inline src/Bot/PlayerbotMgr.cpp]: You shouldnt use IsAddclassBot?

File: src/Bot/PlayerbotAI.h [hunk 3]
```diff
@@ -533,15 +542,18 @@ class PlayerbotAI : public PlayerbotAIBase
...
+    bool IsRandomBot() { return GetAccountType() == BotAccountType::RANDOMBOT; }
+    bool IsAddclass() { return GetAccountType() == BotAccountType::ADDCLASS; }
     // Bot has a master that is a player.
     bool HasRealPlayerMaster();
     // Bot has a master that is activly playing.
     bool HasActivePlayerMaster();
     // Get the group leader or the master of the bot.
     // Checks if the bot is summoned as alt of a player
-    bool IsAlt();
+    bool IsAlt() { return GetAccountType() == BotAccountType::ALT; }
```
Comment (kadeshar) [inline src/Bot/PlayerbotAI.h]: There is possibility that bot will loose alt status in session? Like unlink linked account? That is covered?

File: src/Bot/PlayerbotAI.h [hunk 4]
```diff
@@ -650,6 +662,7 @@ class PlayerbotAI : public PlayerbotAIBase
     Position jumpDestination = Position();
     uint32 nextTransportCheck = 0;
     bool spellInterruptRequested = false;
+    BotAccountType _accountType = BotAccountType::UNDEFINED;
```
Comment (kadeshar) [inline src/Bot/PlayerbotAI.h]: This property initialization for REALPLAYER is confusing. Why in HandlePlayerbotCommand?

Comment (kadeshar) [inline src/Bot/PlayerbotAI.h]: All methods Is... assume that _accountType is initialized. But can be called before that. Shouldnt be error or something like this?

File: src/Bot/PlayerbotAI.h (hunk 3 above)
Comment (Celandriel) [reply src/Bot/PlayerbotAI.h]: Not really. It's only set during bot login, so I don't think there is a way that it would change mid session

File: src/Bot/PlayerbotAI.h (hunk 4 above)
Comment (Celandriel) [reply src/Bot/PlayerbotAI.h]: So in this case it's because it's the best way to identify if a botAI is for a real player. The only way real players have AI is through self bot command.

Comment (Celandriel) [reply src/Bot/PlayerbotAI.h]: It seemed like good practice to have a starting value. Maybe an error can be set in the getter.

File: src/Bot/PlayerbotAI.h (hunk 3 above)
Comment (kadeshar) [reply src/Bot/PlayerbotAI.h]: That not means that will resolve wrong because of that?

**Comment by kadeshar** [comment]
Also that should be PR to test-staging right?

**Review by NoxMax — CHANGES_REQUESTED** [review]

**Comment by Celandriel** [comment]
Did some clean up and extra testing on this. I cant think of any weird edge cases right now.

File: src/Bot/PlayerbotAI.h (hunk 3 above)
Comment (Celandriel) [reply src/Bot/PlayerbotAI.h]: Right now, I dont think that an account that is unlinked would cause a loss of permissions over an altbot you already control, because they would already have you as master, and that behavior isnt changed by this. Really, its just intended to make the typing much explicit.

File: src/Bot/PlayerbotAI.h (hunk 4 above)
Comment (kadeshar) [reply src/Bot/PlayerbotAI.h]: Maybe GetBotType should raise error when called before proper initalization

**Comment by kadeshar** [comment]
Conflict to resolve

**Review by Celandriel — COMMENTED** [review]
> Maybe GetBotType should raise error when called before proper initalization 

Resolved.

**Review by kadeshar — APPROVED** [review]
